### PR TITLE
Improve datetime functions

### PR DESCRIPTION
**Pull request by qgolsteyn**
This PR ensures all dates used are in the PST timezone, and adds associated tests to ensure the behaviour of these functions is correct.

Closes #50 

Would recommend testing it locally to make sure caching is still working as intended before merging.

### CODE REVIEW

**Comment by qgolsteyn** [comment]
Hold off before merging, this is not fully integrated with `parser.js` and `timeseries.js`

**Comment by lazd** [comment]
@qgolsteyn can we do #48 as part of this? For cache as well.

**Comment by hyperknot** [comment]
I opened #194 and I see there is work already here. What I recommend:
- Use [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) standard, as detailed in #194. This is an industry standard human readable data function, with very well tested parsing and writing functions in all date libraries in all programming languages.
- We should not roll our own date methods.
- I recommend Luxon the recommended library from Moment.js developers. Think of it as a modern version of Moment.js.
- We should use UTC dates everywhere.

**Comment by hyperknot** [comment]
Luxon has [.toISODate()](https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#instance-method-toISODate) and [.fromISO()](https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#static-method-fromISO), super well tested methods.

**Comment by lazd** [comment]
> We should not roll our own date methods.
Yeah uhg, I should have started with moment or Luxon, but I was lazy.

**Comment by ryanblock** [comment]
Would like to suggest [Spacetime](https://github.com/spencermountain/spacetime) as an alternative. Super lightweight and pretty flexible. Example impl: https://github.com/covidatlas/covidatlas/blob/master/src/shared/utils/get-date.js

File: package.json [hunk 1]
```diff
@@ -34,5 +36,8 @@
     "@babel/preset-env": "^7.8.7",
     "babel-jest": "^25.1.0",
     "jest": "^25.1.0"
+  },
+  "engines": {
+    "node": ">=13.0.0"
```
Comment (ryanblock) [inline package.json]: Stable Node.js only, please!

Comment (lazd) [reply package.json]: Yeah I think that's leftover from before

**Comment by HerbCaudill** [comment]
Please see my comment [here](https://github.com/lazd/coronadatascraper/issues/42#issuecomment-602115257). Moment and Luxon (and Spacetime AFAICT) all replicate the JS date object's original sin of forcing you to attach times and time zones to your dates, which makes everything much more complicated than it needs to be.

**Comment by hyperknot** [comment]
@ryanblock I don't see what does Spacetime bring over luxon. Luxon is super stable and well tested. In Spacetime I don't even see the basic `fromISO()` and `toISO()` and `toISODate()` methods.

**Comment by qgolsteyn** [comment]
#228 mentioned using [`js-joda`](https://js-joda.github.io/js-joda/manual/LocalDate.html). This library provides a `LocalDate`, which represents a date with no time and no time zone in the ISO-8601 calendar system. It does not use the Date object, does not force you to set a timezone. Could be an interesting option.

**Comment by qgolsteyn** [comment]
Closing, this may not be the right approach. #42 for the up-to-date conversation.

**Pull request closed without merging**